sentry_streams/sentry_streams/k8s/operator/handlers.py

```diff
@@ -0,0 +1,140 @@
+
+        self.core_v1_client = kubernetes.client.CoreV1Api()
+        self.apps_v1_client = kubernetes.client.AppsV1Api()
+
+    def configure(self, logger, **_):
+        logger.info("Running configure method ...")
+        logger.info(f"{self.PIPELINES_CONFIGMAP=}")
+
+        pipelines = self.read_pipelines_configmap()
+        self.apply_pipelines(pipelines)
+
+    def read_pipelines_configmap(self):
+        api_res = self.core_v1_client.read_namespaced_config_map(
+            self.PIPELINES_CONFIGMAP,
+            self.NAMESPACE,
+        )
+        self.logger.info(f"{api_res=}")
+        self.logger.info(f"{api_res.data=}")
+
+        pipelines = {}
+        for filename, config_txt in api_res.data.items():
+            pipelines[filename] = yaml.safe_load(config_txt)
+
+        return pipelines
+
+    def apply_pipelines(self, pipelines):
+        for pipeline_name, pipeline_config in pipelines.items():
+            self.logger.debug(f"Handling {pipeline_name=}")
+            cm = generate_configmap(
+                config=pipeline_config, configmap_template=self.configmap_template
+            )
+            apply_namespace(k8s_resources=[cm], namespace=self.NAMESPACE)
+
+            self.create_or_patch_configmap(body=cm)
+
+            deployments = generate_deployments(
+                config=pipeline_config,
+                deployment_template=self.deployment_template,
+                image=self.image,
+                container_name=self.container_name,
+            )
+            apply_namespace(k8s_resources=deployments, namespace=self.NAMESPACE)
+
+            for deployment in deployments:
+                self.create_or_patch_deployment(body=deployment)
+
+    def create_or_patch_configmap(self, body):
+        name, namespace = body["metadata"]["name"], body["metadata"]["namespace"]
+        try:
+            api_res = self.core_v1_client.read_namespaced_config_map(name=name, namespace=namespace)
+        except K8sApiException as exc:
+            if exc.status == 404:
+                # Create a new configmap.
+                self.logger.info(f"Create config map {name=}")
+                self.core_v1_client.create_namespaced_config_map(namespace=namespace, body=body)
+            else:
+                raise
+        # config-map already exists patch it
+        self.core_v1_client.patch_namespaced_config_map(name=name, namespace=namespace, body=body)
+
+    def create_or_patch_deployment(self, body):
+        name, namespace = body["metadata"]["name"], body["metadata"]["namespace"]
+        try:
+            api_res = self.apps_v1_client.read_namespaced_deployment(name=name, namespace=namespace)
+        except K8sApiException as exc:
+            if exc.status == 404:
+                self.logger.info(f"Create deployment {name=}")
+                self.apps_v1_client.create_namespaced_deployment(namespace=namespace, body=body)
+            else:
+                raise
+        # do not patch the image
```

Comment:
Is this because of GoCD ?

Reply:
Yes, correct, so that the current way we deploy new images keeps working even for deployments managed by the operator.